lightning/src/ln/channel.rs

```diff
@@ -898,7 +912,18 @@ impl<Signer: Sign> Channel<Signer> {
 			// We've exhausted our options
 			return Err(());
 		}
-		self.channel_type = ChannelTypeFeatures::only_static_remote_key(); // We only currently support two types
+		// We support opening a few different types of channels. Try removing our additional
+		// features one by one until we've either arrived at our default or the counterparty has
+		// accepted one.
+		if self.channel_type.supports_anchors_zero_fee_htlc_tx() {
```

Comment:
Do we have test coverage for `handle_error` once we got an error message due to rejection of channel type ?